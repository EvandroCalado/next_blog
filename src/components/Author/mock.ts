export default {
  title: 'Evandro Calado',
  resume: 'Desenvolvedor Frontend',
  image: 'https://avatars.githubusercontent.com/u/110628201?v=4',
  alt: 'Evandro Calado',
};
