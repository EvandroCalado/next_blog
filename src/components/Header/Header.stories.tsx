import Header from './Header';
import { Meta } from '@storybook/react';

export default {
  title: 'Header',
  component: Header,
} as Meta;

export const Template = (args: JSX.IntrinsicAttributes) => (
  <div>
    <Header {...args} />
  </div>
);
