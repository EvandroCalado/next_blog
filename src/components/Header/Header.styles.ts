import styled, { css } from 'styled-components';

export const Container = styled.header`
  ${({ theme }) => css`
    background-color: ${theme.colors.primary};
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.medium};
    padding: ${theme.spacings.medium};
    margin-bottom: ${theme.spacings.large};
    text-align: center;

    a {
      display: flex;
      align-items: center;
      justify-content: center;
      gap: ${theme.spacings.medium};
    }
  `}
`;

export const Title = styled.h1`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.large};
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 10rem;
    border-radius: 50%;
    border: 2px solid ${theme.colors.secondary};
  `}
`;
