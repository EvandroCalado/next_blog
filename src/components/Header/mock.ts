export default {
  image: 'https://avatars.githubusercontent.com/u/110628201?v=4',
  title: 'Evandro Calado',
};
