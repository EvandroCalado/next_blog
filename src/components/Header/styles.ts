import styled, { css } from 'styled-components';
import { Button } from '../ToggleButton/styles';

export const Header = styled.header`
  ${({ theme }) => css`
    width: 100%;
    background-color: ${theme.colors.primary};
    padding: ${theme.spacings.medium} 0;
    margin-bottom: ${theme.spacings.large};
    transition: all 300ms ease-in-out;
  `}
`;

export const Wrapper = styled.div`
${({ theme }) => css`
    position: relative;
    width: 100%;
    max-width: 120rem;
    margin: 0 auto;
    padding: 0 ${theme.spacings.medium};
    display: flex;
    justify-content: space-between;

    ${Button} {
      position: absolute;
      right: ${theme.spacings.medium};
      bottom: 0;
    }
  `}
  
`;
