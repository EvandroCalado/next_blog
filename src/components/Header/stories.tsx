import Header from '.';
import { Meta, StoryFn } from '@storybook/react';
import { HeaderProps } from '.';

import mock from './mock';

export default {
  title: 'Header',
  component: Header,
  args: mock,
} as Meta;

export const Template: StoryFn<HeaderProps> = (args) => (
  <div>
    <Header {...args} />
  </div>
);
