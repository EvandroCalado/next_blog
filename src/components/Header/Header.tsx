import Link from 'next/link';
import * as Styled from './Header.styles';

export type HeaderProps = {
  image: string;
  title: string;
  description: string;
};

const Header = ({ image, title, description }: HeaderProps) => {
  return (
    <Styled.Container>
      <Link href={'/'}>
        <Styled.Image src={image} alt={title} />
      </Link>
      <Styled.Title>{title}</Styled.Title>
      <Styled.Description>{description}</Styled.Description>
    </Styled.Container>
  );
};

export default Header;
