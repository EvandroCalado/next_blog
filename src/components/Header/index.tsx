import * as Styled from './styles';
import Search from '../Search';
import Logo from '../Logo';
import ToggleButton from '../ToggleButton';

export type HeaderProps = {
  image: string;
  title: string;
  description: string;
};

const Header = ({ image, title, description }: HeaderProps) => {
  return (
    <Styled.Header>
      <Styled.Wrapper>
        <Logo image={image} title={title} description={description} />
        <Search />
        <ToggleButton />
      </Styled.Wrapper>
    </Styled.Header>
  );
};

export default Header;
