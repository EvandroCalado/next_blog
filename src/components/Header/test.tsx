import Header from '.';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';

import mock from './mock';

describe('<Header />', () => {
  it('test_logo_receives_props', () => {
    renderTheme(<Header {...mock} />);

    expect(screen.getByAltText(mock.title)).toBeInTheDocument();
    expect(screen.getByText(mock.title)).toBeInTheDocument();
    expect(screen.getByText(mock.description)).toBeInTheDocument();
  });

  it('should_to_match_snapshot', () => {
    const { container } = renderTheme(<Header {...mock} />);

    expect(container).toMatchSnapshot();
  });
});
