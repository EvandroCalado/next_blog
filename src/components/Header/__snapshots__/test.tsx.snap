// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should_to_match_snapshot 1`] = `
.c11 {
  color: #f2f0d8;
  outline: none;
  border: none;
  background-color: transparent;
  cursor: pointer;
}

.c11 svg {
  width: 4rem;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c7 {
  width: 30rem;
  background-color: #f2f0d8;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem 1rem;
  border: 1px solid #252422;
  border-radius: .5rem;
}

.c7 svg {
  width: 3rem;
  margin-right: 0.5rem;
  color: #252422;
}

.c9 {
  width: 100%;
  background-color: #f2f0d8;
  color: #252422;
  font-size: 2rem;
  border: none;
  outline: none;
}

.c9::-webkit-input-placeholder {
  color: #252422;
}

.c9::-moz-placeholder {
  color: #252422;
}

.c9:-ms-input-placeholder {
  color: #252422;
}

.c9::placeholder {
  color: #252422;
}

.c0 {
  width: 100%;
  background-color: #252422;
  padding: 2.4rem 0;
  margin-bottom: 3.2rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c1 {
  position: relative;
  width: 100%;
  max-width: 120rem;
  margin: 0 auto;
  padding: 0 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 .c10 {
  position: absolute;
  right: 2.4rem;
  bottom: 0;
}

.c8 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 2rem;
  color: #f2f0d8;
  font-size: 2.4rem;
}

.c2 a {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 2.4rem;
}

.c3 {
  width: 10rem;
  border-radius: 50%;
  border: 2px solid #f24b4b;
}

.c4 {
  font-size: 3.2rem;
  color: #f24b4b;
}

.c5 {
  font-size: 1.6rem;
  color: #f24b4b;
}

@media (max-width:768px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    gap: 2rem;
  }

  .c1 .c6 {
    width: 100%;
  }

  .c1 .c10 {
    top: 5rem;
    right: 0;
  }

  .c1 .c10 svg {
    width: 3rem;
  }
}

@media (max-width:768px) {
  .c4 {
    font-size: 2.4rem;
  }
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          href="/"
        >
          <img
            alt="Evandro Calado"
            class="c3"
            src="https://avatars.githubusercontent.com/u/110628201?v=4"
          />
        </a>
        <div
          class=""
        >
          <h1
            class="c4"
          >
            Evandro Calado
          </h1>
          <p
            class="c5"
          >
            uma descrição
          </p>
        </div>
      </div>
      <form>
        <div
          class="c6 c7"
        >
          <svg
            aria-hidden="true"
            class="c8"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0 0 16 9.5 6.5 6.5 0 1 0 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <input
            class="c9"
            placeholder="Pesquisa"
            type="search"
            value=""
          />
        </div>
      </form>
      <button
        class="c10 c11"
      >
        <svg
          aria-hidden="true"
          class="c8"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 9c1.65 0 3 1.35 3 3s-1.35 3-3 3-3-1.35-3-3 1.35-3 3-3m0-2c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zM2 13h2c.55 0 1-.45 1-1s-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1zm18 0h2c.55 0 1-.45 1-1s-.45-1-1-1h-2c-.55 0-1 .45-1 1s.45 1 1 1zM11 2v2c0 .55.45 1 1 1s1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1zm0 18v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1zM5.99 4.58a.996.996 0 0 0-1.41 0 .996.996 0 0 0 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0s.39-1.03 0-1.41L5.99 4.58zm12.37 12.37a.996.996 0 0 0-1.41 0 .996.996 0 0 0 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0a.996.996 0 0 0 0-1.41l-1.06-1.06zm1.06-10.96a.996.996 0 0 0 0-1.41.996.996 0 0 0-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06zM7.05 18.36a.996.996 0 0 0 0-1.41.996.996 0 0 0-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06z"
          />
        </svg>
      </button>
    </div>
  </header>
</div>
`;
